File: scrapy-backend/services/storage-service/internal/services/storage_service.go
package services

import (
	"context"
	"storage-service/internal/models"
	"storage-service/internal/repository"
)

type StorageService struct {
	StorageRepo *repository.StorageRepository
}

func NewStorageService(repo *repository.StorageRepository) *StorageService {
	return &StorageService{StorageRepo: repo}
}

func (s *StorageService) StoreSearchResults(ctx context.Context, req *models.SearchResult) error {
	if err := s.StorageRepo.StoreSearchResults(ctx, req); err != nil {
		return err
	}
	return nil
}

func (s *StorageService) GetSearchResults(ctx context.Context, searchID string) (*models.SearchResults, error) {
	return s.StorageRepo.GetSearchResults(ctx, searchID)
}

func (s *StorageService) GetUserSearchResults(ctx context.Context, userID string, page int, pageSize int) (int, []models.UserSearchResults, error) {
	return s.StorageRepo.GetUserSearchResults(ctx, userID, page, pageSize)
}

File: scrapy-frontend/src/app/core/services/backend/storage.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../../../environments/environments';
import { HttpService } from '../http/http.service';
import { Observable } from 'rxjs';
import { StorageApis } from '../../constants/api-routes-constants';
import {
  SearchResultsResponse,
  UserSearchResultsParams,
  UserSearchResultsResponse,
} from '../../../models/storage.model';

@Injectable({
  providedIn: 'root',
})
export class StorageService {
  private readonly apiUrl = environment.api.storage;

  constructor(private http: HttpService) {}

  getSearchResults(searchId: string): Observable<SearchResultsResponse> {
    const url = `${StorageApis.SEARCH_RESULTS}/${encodeURIComponent(searchId)}`;
    return this.http.get<SearchResultsResponse>(url, this.apiUrl);
  }

  getUserSearchResults(
    params: UserSearchResultsParams
  ): Observable<UserSearchResultsResponse> {
    const url = `${StorageApis.USER_SEARCH_RESULTS}/${encodeURIComponent(
      params.userId
    )}?page=${params.page}&page_size=${params.pageSize}`;
    return this.http.get<UserSearchResultsResponse>(url, this.apiUrl);
  }
}
